src/main/kotlin/com/cognifide/gradle/aem/common/instance/service/osgi/OsgiFramework.kt

```diff
@@ -132,6 +134,62 @@ class OsgiFramework(sync: InstanceSync) : InstanceService(sync) {
         enableComponent(pid)
     }
 
+    fun findConfiguration(pid: String): OSGiConfiguration? {
+        return sync.http.get("$CONFIGURATION_PATH/$pid?post=true") { asObjectFromJson(it, OSGiConfiguration::class.java) }
+    }
+
+    fun getConfiguration(pid: String): OSGiConfiguration {
+        aem.logger.debug("Asking for OSGi configuration on $instance")
+        return findConfiguration(pid)
+                ?: throw InstanceException("OSGi Configuration for PID $pid was not found on $instance. Make sure the component exists.")
+    }
+
+    fun getConfigurations(pid: String): List<OSGiConfiguration> {
+        return getConfigList()
+                .filter { pid == it.fpid }
+                .mapNotNull { findConfiguration(it.id) }
+    }
+
+    fun createConfiguration(pid: String, properties: Map<String, Any> = mapOf()) {
+        val currentConfiguration = getConfiguration(pid)
+        val newConfiguration = getConfigPayload(currentConfiguration, properties)
+        sync.http.post("$CONFIGURATION_PATH/$pid", newConfiguration) { response ->
+            aem.logger.debug("Response from instance $instance and pid $pid: ${response.statusLine.statusCode}")
+        }
+        println(getConfigList().size)
+    }
+
+    fun createConfiguration(pid: String, service: String, properties: Map<String, Any> = mapOf()) {
+        createConfiguration("$pid~$service", properties)
+    }
+
+    fun updateConfiguration(pid: String, properties: Map<String, Any> = mapOf()) {
+        val currentConfiguration = getConfiguration(pid)
```

Comment:
is it really needed to calculate a diff? why `getConfiguration` is called? is it really necessary?

why not just `saving` configuration instead of having 2 methods `create` and `update` ?

Reply:
Due to an edge case the diff between old and updated config is necessary. Still, one `save` method is enough. Code updated